examples/demo.rs

```diff
@@ -223,9 +223,36 @@ impl ApplicationHandler<()> for State {
             WindowEvent::RedrawRequested => {
                 profiling::scope!("Redraw Requested");
 
-                let frame = surface
-                    .get_current_texture()
-                    .expect("Failed to acquire next surface texture");
+                let mut frame;
```

Comment:
that's nice but also out of place for this PR
Any reason it does 10x retry without pumping the window loop, has this been observed to be actually getting around issues? (I'm curious!)

Reply:
It failed to acquire sometimes. 10 retires is arbitrary but fixed it for me - Debian 12, wayland.

Have done it this way in other winit applications I have been working on and it seems to work